drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -4635,6 +4636,182 @@ String RenderingDeviceVulkan::_shader_uniform_debug(RID p_shader, int p_set) {
 	return ret;
 }
 
+void RenderingDeviceVulkan::_create_specialization_spirv(const Vector<uint8_t> &p_spirv_data, const Vector<Shader::SpecializationConstant> &p_shader_specialization_constants, RenderingDeviceVulkan::Shader::SpecializationSPIRV &p_spec_spirv) {
```

Comment:
This could be better put in `RenderingDevice`, as sibling of `_reflect_spirv()`. After all, even if we only know we want this for Vulkan, seems like an API-agnostic SPIR-V thing for Godot. Maybe in the future other APIs benefit somehow from providing pre-patched SPIR-V. No big deal in any case, since with the change to `RenderingDeviceDriver` this will end up somewhere else anway.

Reply:
Sounds fair to me, making it API-agnostic is probably for the best. Should we solve this before or after RDD rewrites?

Reply:
The most "correct" way would be moving it now. I'll put it where it belongs when I do the (likely painful) rebase of the RDD work.